src/main/java/core/basesyntax/servise/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.servise;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String REPORT_HEADER = "fruit,quantity" + System.lineSeparator();
+    private final Map<String, Integer> storage;
```

Comment:
Follow the encapsulation principle. The `storage` map should be private and not passed through the constructor. Instead, consider using a service to retrieve the data when needed.